### ISSUE TITLE
Yolo's Data Augmentation for generalized vision models

### ISSUE THREAD

**Issue by GrayMitch** [ultralytics/ultralytics#24454]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

Hi guys, I was wondering if anyone could help me understand something.

I need some help to understand exactly why the data augmentation is a valuable part of the training pipeline for yolo models. It seems to me through repeated testing and development of a multitude of computer vision models that data augmentation leads to the model performing worse in real life. 

I understand that it is used to help make the dataset more unique and to prevent the models from overfitting, however the final product of the training is always significantly worse when I apply the yolo's standard rand augment data augmentation to my data. Given that you can adjust it I see the merit in what it can offer however for a user to fine tune these individual specifications is extremely time consuming and is basically trial and error as there is no standard.

Would anyone have any idea on the best approach to understanding this topic as I think it may be useful to building a more "general vision model"

Thank you!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @GrayMitch, thank you for your interest in Ultralytics 🚀! This is an automated response to help get your question routed quickly, and an Ultralytics engineer will also assist soon.
We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.
If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.
If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).
For your augmentation question specifically, it would be especially helpful if you can share:
- Your model version and `ultralytics` package version
- A summary of your dataset and target deployment conditions
- The augmentation settings you tested
- Training results with and without augmentation
- Any validation metrics, confusion matrices, or example failure cases
Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.
## Upgrade
Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:
```bash
pip install -U ultralytics
```
## Environments
YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):
- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo26"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>
## Status
<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>
If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by Y-T-G** [comment]
Hello.

Are you referring to classification models? Because `randaugment` is only for classification models.

**Comment by GrayMitch** [comment]
Is randaugment not applicable to object detection or pose? If you were to apply an augmentation would you have to manually adjust each individual factor?

**Comment by AffanBinFaisal** [comment]
Hi @GrayMitch

For detection and pose RandAugment is not part of the training path in this repo. It only runs for classification where you get one label for the whole image. Boxes and keypoints use the other stuff like mosaic, color jitter, flips, and so on.

You do not need to tweak RandAugment field by field. For classify you mostly pick the policy name and a few normal training options. For detect and pose you tune the YOLO knobs, not RandAugment.

If real world quality dropped on a detector, RandAugment was probably never doing anything. Worth double checking you meant classify if you are chasing auto augment.

One more thing that trips people up: the big default config still lists auto augment and it often gets printed in the training banner even when your task is detect or pose. So it can look like RandAugment is on when it is not actually used. That is basically noise for those tasks.

**Comment by Y-T-G** [comment]
> Is randaugment not applicable to object detection or pose? If you were to apply an augmentation would you have to manually adjust each individual factor?

The `randaugment` that's available through `auto_augment` argument in Ultralytics is only applicable for classification model. 

So if that's what you're referring to by `randaugment`, then it doesn't have an effect on object detection or pose.

For detection, each augmentation has a separate config that you can adjust.

**Comment by GrayMitch** [comment]
Okay thank you very much!

**Comment by glenn-jocher** [comment]
You’re welcome — our [data augmentation guide](https://docs.ultralytics.com/guides/yolo-data-augmentation) is the best next reference for choosing task-appropriate settings, and if you want help tuning a specific detect or pose setup, please open a [GitHub Discussion](https://github.com/orgs/ultralytics/discussions) with your training config and validation results.
